assets/src/edit-story/components/movable/index.js

```diff
@@ -0,0 +1,141 @@
+/**
+ * External dependencies
+ */
+import Moveable from 'react-moveable';
+import PropTypes from 'prop-types';
+
+/**
+ * WordPress dependencies
+ */
+import { useRef, useEffect } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import { useStory } from '../../app';
+
+const CORNER_HANDLES = [ 'nw', 'ne', 'sw', 'se' ];
+const ALL_HANDLES = [ 'n', 's', 'e', 'w', 'nw', 'ne', 'sw', 'se' ];
+
+function Movable( {
+	rotationAngle,
+	x,
+	y,
+	type,
+	targetEl,
+	pushEvent,
+	setPushEvent,
+} ) {
+	const moveable = useRef();
+
+	const {
+		actions: { setPropertiesOnSelectedElements },
+	} = useStory();
+
+	useEffect( () => {
+		if ( moveable.current ) {
+			// If we have persistent event then let's use that.
+			if ( pushEvent ) {
+				moveable.current.moveable.dragStart( pushEvent );
+				// Remove the event right after using, not to influence other elements.
+				setPushEvent( null );
+			}
+			moveable.current.updateRect();
+		}
+		// Disable reason: we should not run this when pushEvent changes!
+		// eslint-disable-next-line react-hooks/exhaustive-deps
+	}, [ targetEl, moveable ] );
+
+	const frame = {
```

Comment:
We can reconsider this with grouping, but just realized - is it ok that this element is reset on each re-render? Should it be a `ref`?

Reply:
Probably should be reset when re-rendering, or which case do you have in mind?

Reply:
It seemed like re-rendering could happen somewhat randomly while in the middle of dragging. Might be a too obscure case for now, but there are some many dependencies via props, state, and context, that re-rendering is probably pretty chatty.